back/app/models/custom_page.rb

```diff
@@ -49,6 +53,9 @@ class CustomPage < ApplicationRecord
 
   validates :slug, presence: true, uniqueness: true
 
+  validates :code, inclusion: { in: CODES }
+  validates :code, uniqueness: true, if: ->(page) { !page.custom? }
```

Comment:
```suggestion
  validates :code, inclusion: %w[custom]
```
I think this is all that we want here, agree?